### PR TITLE
[WIP] Data Bus

### PR DESCRIPTION
**Pull request by kemitix**
See issue #467

### CODE REVIEW

**Comment by iluwatar** [comment]
@kemitix in general the example looks fine. Here are some things we still need to do before merging:
- Add data-bus as a module to the parent `pom.xml` so it is built with the rest of the patterns
- Leave out Lombok library, we do not use it here.

File: data-bus/pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            <artifactId>lombok</artifactId>
+            <version>${lombok.version}</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (iluwatar) [inline data-bus/pom.xml]: I'm not sure why maven-surefire-plugin is needed? Can we leave it out?

File: data-bus/src/main/java/com/iluwatar/databus/App.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.iluwatar.databus.members.CounterMember;
+import com.iluwatar.databus.members.StatusMember;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.LocalDateTime;
+
+/**
+ * The Data Bus pattern
+ * <p>
+ * <p>{@see http://wiki.c2.com/?DataBusPattern}</p>
```
Comment (iluwatar) [inline data-bus/src/main/java/com/iluwatar/databus/App.java]: Please describe the pattern here. Also add description how the example implements the pattern.

**Comment by iluwatar** [comment]
@kemitix any update on this?

**Comment by kemitix** [comment]
Sorry.  Been caught up in other projects.  Will update this later today.

**Comment by iluwatar** [comment]
@kemitix ready for another review?

**Comment by kemitix** [comment]
@iluwatar Sure.

File: data-bus/pom.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+
+-->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <properties>
+        <lombok.version>1.16.14</lombok.version>
```
Comment (iluwatar) [inline data-bus/pom.xml]: Remove this line since lombok is not used

**Comment by iluwatar** [comment]
@all-contributors please add @kemitix for code

**Pull request merged**